src/main/java/org/killbill/billing/plugin/analytics/api/user/AnalyticsUserApi.java

```diff
@@ -114,4 +128,43 @@ public void rebuildAnalyticsForAccount(final UUID accountId, final CallContext c
         allBusinessObjectsDao.update(businessContextFactory);
         logger.info("Finished Analytics refresh for account {}", businessContextFactory.getAccountId());
     }
+
+    public RefreshResult rebuildAnalyticsForAllAccounts(final CallContext context) {
+        logger.info("Starting Analytics refresh for all accounts ");
+        Pagination<Account> accounts = osgiKillbillAPI.getAccountUserApi().getAccounts(0L, 100L, context);
+        logger.info("Total accounts to be refreshed {} ",accounts.getTotalNbRecords());
+        RefreshResult refreshResult = rebuildAnalyticsForAccounts(accounts, new RefreshResult(accounts.getTotalNbRecords(), 0L), context);
+        Long nextOffSet = accounts.getNextOffset();
+        while (nextOffSet != null) {
+            accounts = osgiKillbillAPI.getAccountUserApi().getAccounts(nextOffSet, 100L, context);
+            refreshResult = rebuildAnalyticsForAccounts(accounts, refreshResult, context);
+            nextOffSet = accounts.getNextOffset();
+        }
+
+        logger.info("Finished Analytics refresh for all accounts");
+        return refreshResult;
+        
+    }
+
+    private RefreshResult rebuildAnalyticsForAccounts(final Pagination<Account> accounts, RefreshResult refreshResult, final CallContext context) {
+        final Iterator<Account> accountsItr = accounts.iterator();
+        Long nbRefreshes = refreshResult.getNbRefreshes();
+        try {
+            while (accountsItr.hasNext()) {
+                final UUID accountId = accountsItr.next().getId();
+                final AnalyticsJob analyticsJob = new AnalyticsJob(Group.ALL, ExtBusEventType.ACCOUNT_CHANGE, ObjectType.ACCOUNT, null, accountId, context.getTenantId());
+                analyticsListener.scheduleAnalyticsJob(analyticsJob, analyticsConfigurationHandler.getConfigurable(context.getTenantId()));
```

Comment:
Here, I've specified the event type as `ACCOUNT_CHANGE`. Is this correct?

Reply:
I think it would be cleaner to pass something else (e.g. `null`, but not sure it's supported, or maybe even introduce a new `boolean manualRefresh` field in the `AnalyticsJob` POJO), so that it's clearer that these entries were introduced as part of a manual refresh (and not triggered by the system). This could be done in a second patch though, happy to merge as-is for now.